### PR TITLE
Add recommended labels for each components

### LINKED ISSUE #1471: Use Standard Helm Labels/Annotations
This Helm chart is not compliant with Helm Label and Annotations.

## Recommendation 
`helm.sh/chart: NAME-VERSION` as a label so that operators can conveniently find all the instances of a particular chart to use.

## Standard Labels

Name | Status | Description
-- | -- | --
app.kubernetes.io/name | REC | This should be the app name, reflecting the entire app. Usually {{ template "name" . }} is used for this. This is used by many Kubernetes manifests, and is not Helm-specific.
helm.sh/chart | REC | This should be the chart name and version: {{ .Chart.Name }}-{{ .Chart.Version \| replace "+" "_" }}.
app.kubernetes.io/managed-by | REC | This should always be set to {{ .Release.Service }}. It is for finding all things managed by Helm.
app.kubernetes.io/instance | REC | This should be the {{ .Release.Name }}. It aids in differentiating between different instances of the same application.
app.kubernetes.io/version | OPT | The version of the app and can be set to {{ .Chart.AppVersion }}.
app.kubernetes.io/component | OPT | This is a common label for marking the different roles that pieces may play in an application. For example, app.kubernetes.io/component: frontend.
app.kubernetes.io/part-of | OPT | When multiple charts or pieces of software are used together to make one application. For example, application software and a database to produce a website. This can be set to the top level application being supported.

https://helm.sh/docs/chart_best_practices/labels/#standard-labels

https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/

### PR DESCRIPTION
**Pull request by JhoLee**
Add recommended labels for each components of Harbor, and common recommended labels by _helpers.tpl

Reference: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/
 
Resolves #1471 #1045

### CODE REVIEW

File: templates/core/core-dpl.yaml [hunk 1]
```diff
@@ -4,18 +4,18 @@ metadata:
   name: {{ template "harbor.core" . }}
   labels:
 {{ include "harbor.labels" . | indent 4 }}
-    component: core
+{{ include "harbor.labels" . | indent 4 }}
```
Comment (dioguerra) [inline templates/core/core-dpl.yaml:7]: duplicated?

Comment (JhoLee) [reply templates/core/core-dpl.yaml:7]: Oops, typo. It had to be:
```suggestion
{{ include "harbor.core.labels" . | indent 4 }}
```

Comment (JhoLee) [reply templates/core/core-dpl.yaml]: Fixed it.

**Comment by JhoLee** [comment]
Any updates?

**Comment by sidewinder12s** [comment]
FYI, Changing selector labels will require forced redeployment of all components as that field is immutable, so it'll be a semi-breaking change. Might want to look at not doing that.

**Comment by zyyw** [comment]
could you please squash the commits first and then rebase the code as well. @JhoLee

**Comment by zyyw** [comment]
@sidewinder12s why would it be a concern since this PR is backward compatiable.

**Comment by sidewinder12s** [comment]
@zyyw As far as I'd seen, you cannot change the labels used in selectors after a deployment, it'll error out saying its a immutable field. So if this lands, users cannot deploy it without either force applying it or tearing down their existing deployments. 

https://stackoverflow.com/questions/67679922/how-to-remove-a-label-selector-from-a-kubernetes-deployment

**Pull request closed without merging**